api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/HivCohortQueries.java

```diff
@@ -191,6 +197,175 @@ public CohortDefinition getPatientsInArtWhoAbandoned() {
+            + "       AND most_recent.value_datetime < Date_add(:onOrBefore, interval 30 day) "
+            + "       AND most_recent.location_id = :location "
+            + "GROUP  BY most_recent.patient_id ";
+
+    definition.setQuery(
+        String.format(
+            query,
+            hivMetadata.getARVPharmaciaEncounterType().getEncounterTypeId(),
+            hivMetadata.getReturnVisitDateForArvDrugConcept().getConceptId(),
+            hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId(),
+            hivMetadata.getARVPediatriaSeguimentoEncounterType().getEncounterTypeId(),
+            commonMetadata.getReturnVisitDateConcept().getConceptId(),
+            hivMetadata.getArtDatePickup().getConceptId(),
+            hivMetadata.getMasterCardDrugPickupEncounterType().getEncounterTypeId()));
+
+    return definition;
+  }
+
+  /**
+   * 14. All patients who do not have the next scheduled drug pick up date (Fila) and next scheduled
+   * consultation date (Ficha de Seguimento or Ficha Clinica – Master Card) and ART Pickup date
+   * (Recepção – Levantou ARV).
+   *
+   * @return
+   */
+  @DocumentedDefinition(value = "patientWithoutScheduledDrugPickupDateMasterCardAmdArtPickup")
+  private CohortDefinition getPatientWithoutScheduledDrugPickupDateMasterCardAmdArtPickup() {
+    SqlCohortDefinition definition = new SqlCohortDefinition();
+
+    definition.setName("patientWithoutScheduledDrugPickupDateMasterCardAmdArtPickup");
+
+    definition.addParameter(new Parameter("location", "location", Location.class));
+
+    String query =
+        "SELECT p.patient_id "
+            + "FROM   patient p "
+            + "       inner join encounter e "
+            + "               ON e.patient_id = p.patient_id "
+            + "       inner join obs o "
+            + "               ON o.encounter_id = e.encounter_id "
+            + "WHERE  p.voided = 0 "
+            + "       AND e.voided = 0 "
+            + "       AND o.voided = 0 "
+            + "       AND e.encounter_type NOT IN ( %s, %s, %s, %s ) "
+            + "       AND e.location_id = :location "
+            + "GROUP  BY p.patient_id "
+            + "UNION "
+            + "SELECT p.patient_id "
+            + "FROM   patient p "
+            + "       inner join encounter e "
+            + "               ON e.patient_id = p.patient_id "
+            + "       inner join obs o "
+            + "               ON o.encounter_id = e.encounter_id "
+            + "WHERE  p.voided = 0 "
+            + "       AND e.voided = 0 "
+            + "       AND o.voided = 0 "
+            + "       AND e.encounter_datetime = (SELECT Max(encounter_datetime) "
+            + "                                   FROM   encounter "
+            + "                                   WHERE  patient_id = p.patient_id) "
+            + "       AND e.encounter_type IN ( %s, %s, %s ) "
+            + "       AND ( o.concept_id NOT IN ( %s, %s ) "
```

Comment:
@ynurmahomed the specs says: "...or its contents are null"

shouldn't we rather be checking the obs value_coded or value_numeric is null?

Reply:
Re-using from TX_CURR.